packages/frontend/app/routes/auth.recovery.tsx

```diff
@@ -0,0 +1,127 @@
+import { json, type LoaderFunctionArgs } from '@remix-run/node'
+import { uuidSchema } from '~/lib/validate.server'
+import { isUiNodeInputAttributes } from '@ory/integrations/ui'
+import type { UiContainer } from '@ory/client'
+import { useLoaderData } from '@remix-run/react'
+import { Button, Input } from '../components/ui'
+import variables from '../lib/envConfig.server'
+import { redirectIfAlreadyAuthorized } from '../lib/kratos_checks.server'
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const cookies = request.headers.get('cookie')
+  await redirectIfAlreadyAuthorized(request.url, cookies, '/settings')
+
+  const url = new URL(request.url)
+
+  const flowResult = uuidSchema.safeParse({ id: url.searchParams.get('flow') })
+  if (flowResult.success) {
+    const flowId = flowResult.data.id
+    const response = await fetch(
+      `${variables.kratosContainerPublicUrl}/self-service/recovery/flows?id=${flowId}`,
+      {
+        headers: {
+          Cookie: cookies || ''
+        },
+        credentials: 'include'
+      }
+    )
+    if (!response.ok) {
+      throw json(null, {
+        status: 400,
+        statusText: 'Could not fetch Kratos account recovery fields.'
+      })
+    }
+    const responseData = await response.json()
+
+    return { responseData }
+  } else {
+    throw json(null, {
+      status: 400,
+      statusText: 'No Kratos account recovery flow ID found.'
+    })
+  }
+}
+
+export default function Recovery() {
+  const { responseData } = useLoaderData<typeof loader>()
+  const uiContainer: UiContainer = responseData.ui
+  const uiNodes = uiContainer.nodes
+  const actionUrl = uiContainer.action
```

Comment:
out of curiosity, how do these work exactly? the `/self-service/recovery/flows` return a list of UI elements? for a given flow?

Reply:
Sorry, I missed this comment. Kratos allows you to render all your own UIs so their API calls return elements needed on your page and then it's up to you to group them, place them and style them as you see fit. 
It uses the identity schema you defined so it knows what fields are required when and also what email to use for recovery etc. 

Do you think it would be better to include a comment about this somewhere? Or add a point on this in the docs?